src/extra/AutoUi/Collection/Lenses.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { AutoUIContext, BaseResource } from '../schemaOps';
+import { faListUl } from '@fortawesome/free-solid-svg-icons/faListUl';
+import { faMap } from '@fortawesome/free-solid-svg-icons/faMap';
+import { faTh } from '@fortawesome/free-solid-svg-icons/faTh';
+import { ButtonGroup } from '../../../components/ButtonGroup';
+import { Button } from '../../../components/Button';
+
+export enum CollectionLenses {
+	Table = 'table',
+	Grid = 'grid',
+	Map = 'map',
+}
+
+const lensIcons = {
+	[CollectionLenses.Table]: faListUl,
+	[CollectionLenses.Grid]: faTh,
+	[CollectionLenses.Map]: faMap,
+};
+
+interface LensesProps<T> {
+	lens: CollectionLenses;
+	changeLens: (lens: CollectionLenses) => void;
+	autouiContext: AutoUIContext<T>;
+}
+
+export const Lenses = <T extends BaseResource<T>>({
+	autouiContext,
+	lens,
+	changeLens,
+}: LensesProps<T>) => {
+	const supportedLenses = [CollectionLenses.Table];
+
+	if (autouiContext.cardRenderer) {
+		supportedLenses.push(CollectionLenses.Grid);
+	}
+
+	if (autouiContext.geolocation) {
```

Comment:
Do you mean the `geolocation` name ? If so I would leave it like this since this is not a boolean but an obj with longitude and latitude

Reply:
Since that's not implemented yet afaict (we only add the icon in the lenses), how about adding in in a separate PR when we are ready?
On the other hand I'm fine with this since I guess we could just add a `renderer` property in the `geolocation` object or something...